src/utils/options.py

```diff
@@ -0,0 +1,313 @@
+        self.log_level = raw_config["log_level"]
+        self.log_console_bool = raw_config["log_handlers"]["console"]
+        self.log_logfile_bool = raw_config["log_handlers"]["logfile"]
+        self.log_aggregator_bool = raw_config["log_handlers"]["aggregator"]
+        self.log_aggregator_addr = raw_config["log_aggregator_addr"]
+        self.log_aggregator_port = int(raw_config["log_aggregator_port"])
+        self.hdw_path = raw_config['hdw_path']
+
+    def parse_hdw(self):
+        # Load information from the hardware file
+        hdw_dat_file = f'{self.hdw_path}/hdw.dat.{os.environ["RADAR_ID"]}'
+
+        try:
+            with open(hdw_dat_file) as hdwdata:
+                lines = hdwdata.readlines()
+        except IOError:
+            errmsg = f'Cannot open hdw.dat file at {hdw_dat_file}'
+            raise ValueError(errmsg)
+
+        lines[:] = [line for line in lines if line[0] != "#"]  # remove comments
+        lines[:] = [line for line in lines if len(line.split()) != 0]  # remove blanks
+
+        # Take the final line
+        try:
+            hdw = lines[-1]
+        except IndexError:
+            errmsg = f'Cannot find any valid lines in the hardware file: {hdw_dat_file}'
+            raise ValueError(errmsg)
+        # we now have the correct line of data.
+
+        params = hdw.split()
+        if len(params) != 22:
+            errmsg = f'Found {len(params)} parameters in hardware file, expected 22'
+            raise ValueError(errmsg)
+
+        self.status = params[1]  # 1 operational, -1 offline
+        self.geo_lat = params[4]  # decimal degrees, S = negative
+        self.geo_long = params[5]  # decimal degrees, W = negative
+        self.altitude = params[6]  # metres
+        self.boresight = params[7]  # degrees from geographic north, CCW = negative.
+        self.boresight_shift = params[8]  # degrees from physical boresight. nominal 0.0 degrees
+        self.beam_sep = params[9]  # degrees, nominal 3.24 degrees
+        self.velocity_sign = params[10]  # +1.0 or -1.0
+        self.phase_sign = params[11]  # +1 indicates correct interferometry phase, -1 indicates 180
+        self.tdiff_a = params[12]  # us for channel A.
+        self.tdiff_b = params[13]  # us for channel B.
+        # interferometer offset from midpoint of main, metres [x, y, z] where x is along line of
+        # antennas, y is along array normal and z is altitude difference, in m.
+        self.intf_offset = [float(params[14]), float(params[15]), float(params[16])]
+        self.analog_rx_rise = params[17]  # us
+        self.analog_rx_attenuator = params[18]  # dB
+        self.analog_atten_stages = params[19]  # number of stages
+        self.max_range_gates = params[20]
+        self.max_beams = params[21]  # so a beam number always points in a certain direction
+
+    def parse_restrict(self):
+        # Read in restrict.dat
+        if not os.environ["BOREALISPATH"]:
```

Comment:
Could probably take this repeated code out and put it once at the constructor of the dataclass? Is that possible?